R/sensitivity-tornado-plot.R

```diff
@@ -0,0 +1,304 @@
+  )
+  # override only default configuration values with settings for tornado plot
+  customPlotConfiguration <- defaultPlotConfiguration$clone()
+  customPlotConfiguration <- .updatePlotConfiguration(
+    customPlotConfiguration, tornadoPlotConfiguration
+  )
+
+  # validate plot configuration for valid options
+  plotConfigurationList <- purrr::map(
+    purrr::set_names(names(customPlotConfiguration)),
+    ~ customPlotConfiguration[[.]]
+  )
+  ospsuite.utils::validateIsOption(
+    plotConfigurationList,
+    .getPlotConfigurationOptions(names(tornadoPlotConfiguration))
+  )
+
+  # extract and prepare data -----------------
+
+  data <- .filterPlottingData(
+    data,
+    outputPaths = outputPaths,
+    parameterPaths = parameterPaths,
+    pkParameters = pkParameters
+  )
+
+  # ordered levels for parameter plotting
+  data <- dplyr::group_by(
+    data,
+    OutputPath, ParameterPath, PKParameter, ParameterFactor
+  )
+  data <- dplyr::mutate(
+    data,
+    MeanPercentChangePK = mean(PercentChangePK, na.rm = TRUE)
+  )
+  data <- dplyr::ungroup(data)
+  data <- dplyr::arrange(data, dplyr::desc(abs(MeanPercentChangePK)))
+  data <- dplyr::mutate(
+    data,
+    ParameterPath = purrr::map_chr(ParameterPath, .splitParameterName)
+  )
+  data$ParameterPath <- factor(data$ParameterPath,
+    levels = rev(unique(data$ParameterPath))
+  )
+
+  data <- dplyr::filter(
+    data,
+    ParameterFactor %in% c(parameterFactor, 1 / parameterFactor)
+  )
+
+  # list of plots ----------------------------
+
+  # create plot for each output path
+  lsPlots <- purrr::map(
+    .x = data %>% split(.$OutputPath),
+    .f = ~ .createTornadoPlot(
+      .x,
+      defaultPlotConfiguration = customPlotConfiguration
+    )
+  )
+
+  # print plots without producing warnings
+  suppressWarnings(purrr::walk(lsPlots, ~ print(.x)))
+}
+
+#' @keywords internal
+#' @noRd
+.createTornadoPlot <- function(data,
+                               defaultPlotConfiguration) {
+  # update data dependent plot configuration
+  plotConfiguration <- defaultPlotConfiguration$clone()
+  plotConfiguration <- .updatePlotConfiguration(
+    plotConfiguration,
+    list(title = unique(data$OutputPath))
+  )
+
+  # calculate x-axis breaks and limits -------
+  pLimits <- .calculateLimits(data$PercentChangePK)
+  if (isTRUE(pLimits[1] > 0.5)) pLimits[1] <- 0.5
+  if (isTRUE(pLimits[2] < 1.5)) pLimits[2] <- 1.5
+
+  pBreaks <- .calculateBreaks(data$PercentChangePK, m = 5)
+
+  # map each PK parameter to its own plot ----
+
+  plotList <- purrr::map(
+    unique(data$PKParameter),
+    ~ {
+      dataSubset <- dplyr::filter(data, PKParameter == .x)
+
+      # basic plot setup -------------------------
+
+      plot <- ggplot(
+        dataSubset,
+        aes(
+          x = ParameterPath,
+          y = PercentChangePK,
+          fill = as.factor(ParameterFactor)
+        )
+      ) +
+        geom_col(
+          color = "grey",
+          na.rm = TRUE
+        ) +
```

Comment:
add fixed `width` in geom_col

Reply:
Set width = 0.9. However, this will not control the absolute widths of bars across different plots, as these are influenced by the overall dimensions of the graphics device and the plot layout settings.